namecheap/provider.go

```diff
@@ -1,109 +1,66 @@
-package namecheap
+package namecheap_provider
 
 import (
-	"fmt"
-	"log"
-	"os"
-	"strings"
-	"time"
-
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/namecheap/go-namecheap-sdk/v2/namecheap"
 )
 
-var (
-	errTooManyRetries = fmt.Errorf("exceeded max retry limit")
-)
-
-// These are the "Auto" TTL settings in Namecheap
-const (
-	ncDefaultTTL     int           = 1799
-	ncDefaultMXPref  int           = 10
-	ncDefaultTimeout time.Duration = 30
-)
-
-// Provider returns a terraform.ResourceProvider.
 func Provider() *schema.Provider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
-			"username": {
+			"user_name": {
 				Type:        schema.TypeString,
 				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("NAMECHEAP_USERNAME", nil),
-				Description: "A registered username for namecheap",
+				Description: "A registered user name for namecheap",
 			},
 
 			"api_user": {
 				Type:        schema.TypeString,
 				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("NAMECHEAP_API_USER", nil),
-				Description: "A registered apiuser for namecheap",
+				Description: "A registered api user for namecheap",
 			},
 
-			"token": {
+			"api_key": {
 				Type:        schema.TypeString,
 				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("NAMECHEAP_TOKEN", nil),
```

Comment:
I think we should support passing credentials as env variables

Reply:
Got it.
Reverted back